### PR TITLE
Update nbody example for Rust

### PR DESCRIPTION
**Pull request by MaxGraey**
Relate to #527

### CODE REVIEW

File: examples/n-body/build/untouched.wat [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
- (type $FUNCSIG$iiddddddd (func (param i32 f64 f64 f64 f64 f64 f64 f64) (result i32)))
- (type $FUNCSIG$viiii (func (param i32 i32 i32 i32)))
- (type $FUNCSIG$viii (func (param i32 i32 i32)))
- (type $FUNCSIG$d (func (result f64)))
- (type $FUNCSIG$vid (func (param i32 f64)))
- (type $FUNCSIG$di (func (param i32) (result f64)))
- (type $FUNCSIG$vi (func (param i32)))
+ (type $_ (func))
+ (type $iii (func (param i32 i32) (result i32)))
+ (type $iFFFi (func (param i32 f64 f64 f64) (result i32)))
```
Comment (dcodeIO) [inline examples/n-body/build/untouched.wat]: Looks like this uses dist files / an older version than master

File: examples/n-body/README.md [hunk 2]
```diff
@@ -32,16 +32,18 @@ Benchmark
...
 |        Target           |  Time, ***ms*** | Size, ***KB*** |
 |-------------------------|-----------------|----------------|
-| **AssemblyScript WASM** | **2901**        | **2**          |
-| AssemblyScript ASMJS    | 3720            | 19*            |
-| JavaScript              | 2716            | 5*             |
-| Rust WASM               | 2883            | 13             |
+| **AssemblyScript WASM** | **2921**        | **2**          |
+| AssemblyScript ASMJS    | 3807            | 19*            |
+| JavaScript              | 2757            | 5*             |
+| Rust WASM               | 2866            | 20 / 2**       |
```
Comment (Hywan) [inline examples/n-body/README.md]: I get 13Kb before running `wasm-gc`. And get 1.4Kb after `wasm-gc`. How did you get those numbers?

Comment (MaxGraey) [reply examples/n-body/README.md]: You used `s`  flag instead `3`. I commented about this in prev PR. You could recheck size by yourself using latest nightly

**Comment by MaxGraey** [comment]
> Numbers in the result table aren't correct.

Why? You got another values for sizes?

**Comment by Hywan** [comment]
With the optimisation level `3`, I get the same numbers.

**Comment by MaxGraey** [comment]
Great! I opened new PR because I should update timings anyway (previous measured on my loacal machine)

File: examples/n-body/README.md (hunk 2 above)
Comment (MaxGraey) [reply examples/n-body/README.md]: The main goal of this example is speed bench and size was measured as side effect, so both compilers using `-O3` opt flags

**Pull request merged**